### DISCUSSION TITLE
Bikeshedding down to the name, of all things

### DISCUSSION THREAD

**Discussion by jonct** [Jip-Hop/jailmaker#232] [Ideas]
I'm working on actual contributions — honest!  🤨  But I'll throw this out there just to *get it off of my brain*.

With [reduced focus](https://github.com/Jip-Hop/jailmaker/commit/5a29d4ad2667e4928f02af41f4472a5a00afb672) on the guided wizard feature: the tool might be more aptly described as a "Jailer" than a "Jailmaker".

And as such: `jailer` vs `jlmkr` might be nicer to discuss, and easier to type without injury. 🦾

That is all. Carry on. 🙄

**Comment by mrstux** [comment]
As far as I’m concerned, it’s named, it’s well known by this name, and it’s fairly accurately named, it makes jails, it’s a jailmaker. 

If you wanted to fork, you could rename

Meanwhile, I noticed a switch from referring to jlmkr as a “script” to an “app”. It’s not really an app, and certainly not in the TrueNAS context… what about using “tool”?

**Reply by jonct** [reply]
Great answer. 👍

For sake of discussion: it's well-known by "Jailmaker". But so far it's been invoked as `jlmkr.py`. How strong is the name recognition on that level, and would *\[forgive me\]* brand affinity go up or down if Jailmaker were to get delivered as `jlr` or `jailr` or `jm` vs `jlmkr`?

And yes I agree: definitely "tool" if not "script" or "executable". I'll need to be reminded on where you've picked up "app" — maybe through reference to the `zipapp` module which constructs a `.pyz`-style tool?

**Reply by jonct** [reply]
Future feature proposal; not today but it's relevant to the above: I'd like to see some amount of `argv[0]` "busyboxing" such that if you call it as `jls` or as `jexec` then it behaves as shortcuts for list and exec, etc. Which not so coincidentally play into FreeBSD muscle memory.

**Reply by mrstux** [reply]
I was looking at JipHop's "No Hatch" commit...

https://github.com/Jip-Hop/jailmaker/blob/v3.0.0/README.md#summary

"this app"

And again re `jls` or `jexec`... you are venturing way out of what is necessary, and can simply be achieved with a shell alias... are you not?

The existing guides use "jlmkr" via shell alias... if there's a tool, or if there's a shell alias doesn't matter.

shortcuts could be implemented in the readme "Hey, copy this to your bashrc/zshrc if you'd like"

I'm not sure I understand the desire to grow/complicate Jailmaker... it was/is a simple one file script, that gets run at startup... and used imperatively via a shell alias.

**Reply by jonct** [reply]
> I was looking at JipHop's "No Hatch" commit

Gotcha. Well, I'll take the blame for that anyway. I got a bit cute naming the build steps for a "zipapp" and an "appzip".

> re `jls` or `jexec`... you are venturing way out of what is necessary, and can simply be achieved with a shell alias... are you not?

Maybe elsewhere? This one just feels like a trivial convenience that doesn't get in anyone's way.

> I'm not sure I understand the desire to grow/complicate Jailmaker

I'd like to see it community maintainable. I think it's on that trajectory.

I'd like to see it fulfill the promise of "TrueNAS Core jails, but on TrueNAS SCALE". I think it currently misses that mark, but not by much.

I'm not here to complain about it. I'm offering help.
